pkg/fanal/analyzer/os/release/release.go

```diff
@@ -17,12 +17,43 @@ func init() {
 
 const version = 1
 
-var requiredFiles = []string{
-	"etc/os-release",
-	"usr/lib/os-release",
-	"aarch64-bottlerocket-linux-gnu/sys-root/usr/lib/os-release",
-	"x86_64-bottlerocket-linux-gnu/sys-root/usr/lib/os-release",
-}
+var (
+	requiredFiles = []string{
+		"etc/os-release",
+		"usr/lib/os-release",
+		"aarch64-bottlerocket-linux-gnu/sys-root/usr/lib/os-release",
+		"x86_64-bottlerocket-linux-gnu/sys-root/usr/lib/os-release",
+	}
+
+	idToOSFamilyMapping = map[string]types.OSType{
+		"rhel":                types.RedHat,
+		"centos":              types.CentOS,
+		"rocky":               types.Rocky,
+		"almalinux":           types.Alma,
+		"ol":                  types.Oracle,
+		"fedora":              types.Fedora,
+		"alpine":              types.Alpine,
+		"bottlerocket":        types.Bottlerocket,
+		"opensuse-tumbleweed": types.OpenSUSETumbleweed,
+		"opensuse-leap":       types.OpenSUSELeap,
+		"opensuse":            types.OpenSUSELeap, // opensuse for leap:42, opensuse-leap for leap:15
+		"sles":                types.SLES,
+		// There are various rebrands of SLE Micro, there is also one brief (and reverted rebrand)
+		// for SLE Micro 6.0. which was called "SL Micro 6.0" until very short before release
+		// and there is a "SLE Micro for Rancher" rebrand, which is used by SUSEs K8S based offerings.
+		"sle-micro":         types.SLEMicro,
+		"sl-micro":          types.SLEMicro,
+		"sle-micro-rancher": types.SLEMicro,
```

Comment:
I'm wondering if there is any advantage to using a map here. Isn't switch simpler?

```suggestion
		switch "sle-micro", "sl-micro", "sle-micro-rancher": 
		    return types.SLEMicro
```

Reply:
I was also thinking about whether we should switch to a map.
After adding the RedHat-based distros, this `switch` became very large.
I thought a map would be more readable (even though several keys have the same value, since they are grouped together, it shouldn’t be a problem for readability).

But it’s not critical for me — we can also revert to the switch.

Reply:
> I'm wondering if there is any advantage to using a map here. Isn't switch simpler?

With `switch` condition, the cyclomatic complexity of the function was exceeding the threshold of 20 which was causing one of the linter `gocyclo` failure. to fix it, i changed the `switch` condition into map.

Reply:
`gocyclo` is only for reference, so you may use nolint if you have a convincing reason.

And regarding readability, I think it comes down to personal preference, so since the same values can be grouped together, I personally prefer `switch` this time, but either is fine. Performance-wise, `switch` has [the advantage](https://adayinthelifeof.nl/2021/03/04/go-map-vs-switch.html), but the difference isn’t that big. If you ever want to insert custom trim processing or the like, `switch` might be more convenient. However, since it was originally written with `switch`, I was wondering if there’s really a sufficient reason to go out of the way to change it to a `map`.